### DISCUSSION TITLE
Cannot find module 'date-fns'

### DISCUSSION THREAD

**Discussion by MyklKelsey** [SilentVoid13/Templater#1041] [Help]
When I use a Folder Template a message pops up that says to look in the console. I figured out how to do that and I can see the error:

```
plugin:templater-obsidian:61 Templater Error: Template parsing error, aborting. 
 Cannot find module 'date-fns'
Require stack:
- electron/js2c/renderer_init
```

Does this 'date-fns' need to be installed for Templater to work?

Thanks for your help!

**Comment by welpdx** [comment]
Would be helpful to you have your full code. Sometimes, errors in console point away from the root cause.
Thanks!

**Reply by MyklKelsey** [reply]
Thank you for getting back with me! What is my full code?
I have a Folder Template so that when I create a new file in a folder it should use the template. The only thing in the template is this string: "Starting Simple" That's it.
And I get that date-fns error, which makes me think something isn't configured, some library isn't loaded, or something like that is missing.

I'd be happy to provide more info, just let me know what you need. Is there an even simpler test I could run to troubleshoot?

**Comment by MyklK3** [comment]
I got some help over at the Discord and they said this file is what you need to see...
[obsidian.md-1677599887381.log](https://github.com/SilentVoid13/Templater/files/10852436/obsidian.md-1677599887381.log)

**Comment by Zachatoo** [comment]
What's your installer version of Obsidian? You can check your installer version in settings.

<img width="1123" alt="image" src="https://user-images.githubusercontent.com/6936914/222642336-48294f60-16b3-4289-80e4-b954eda2abb4.png">

I have a wild guess here, but I'm guessing that Templater's "error wrapper" is unintentionally catching an Electron error, and that installing Obsidian from the Obsidian's website over your current install will resolve the issue.

**Reply by Zachatoo** [reply]
Someone else with your same issue in the Obsidian discord with the same issue tried updating their installer version and with a new vault with just the Templater plugin and their issue was not resolved. May be worth trying still.

Link to part of conversation in Discord: <https://discord.com/channels/686053708261228577/875720842443649045/1081095439492853890>

**Reply by MyklKelsey** [reply]
OK, we're getting somewhere!!
I was on installer 1.0.0 and now I'm on installer 1.1.16 ... Boo. It's still broken in my vault.

HOWEVER

I've tested software before... so I did what I know to do. Create the simplest example to eliminate other possibilities.
1. Created a brand new vault with nothing in it but a 'Template' folder and a 'NewFile' folder
2. I loaded Templater
3. I set the location of my templates in the 'Template' folder
4. I enabled "Trigger Template on new file creation
5. I set the Enable Folder Templates on
6. I added 'NewFiles' as the folder using the only template in 'Template'
7. My template page only contains the string 'This template is used for new folder level pages' << no JavaScript
8. Then I replaced my string in step 7. with the "Quick Example" from the Templater GitHub

In my old vault this would trigger the 'date-fns' error. In this new stripped down vault IT DOES NOT.

So now I just need to figure out the difference between these Vaults. (or just move all my files into a new vault??)

**Comment by MyklKelsey** [comment]
OK,
So I had created a new vault, stripped down and only had my one script to run when I created a new page inside a folder. And it worked! So I was up to date on versions. (That's my celebration above)

So I went to my old vault and moved the md files under my new vault and also started messing around with a more complicated starting script for a different folder...

aaand this 'date-fns' is back. So there is something that is causing it. I am going to try this again - go back to a stripped down vault and try to build it up again.

Any help you could give me by looking into what 'date-fns' does. The error also mentions electron:
 Cannot find module 'date-fns'
Require stack:
- electron/js2c/renderer_init
log_error @ plugin:templater-obsidian:61

I know obsidian:61 is just where the error bubbles up from. I hoped these clues might help somebody with more skill and smarts than I have to figure out exactly what's happening.

I will keep digging on this, but any help you could offer would be greatly appreciated.
